Services/DataService.cs

```diff
@@ -0,0 +1,56 @@
+using System.Net.Http.Json;
+using Empower.Models;
+
+namespace DotNetMaui.Services;
+
+public interface IDataService
+{
+    Task Checkout();
+
+    // no quantity - Bruno didn't like data models on the server :)
+    Task AddToCart(Product item);
+    Task RemoveFromCart(Product item);
+    Task<List<Product>> GetCartItems();
+    Task<List<Product>> GetProducts();
+}
+
+public class DataService : IDataService
+{
+    readonly HttpClient httpClient = new(new SentryHttpMessageHandler())
+    {
+        BaseAddress = new(Configuration.BackendUrl)
+    };
+    readonly List<Product> cart = new();
+    
+
+    public async Task Checkout()
+    {
+        await this.httpClient.PostAsync("/checkout", new StringContent(String.Empty)).ConfigureAwait(false);
+        this.cart.Clear();
+    }
+
+    public Task AddToCart(Product item)
+    {
+        ArgumentNullException.ThrowIfNull(item);
+        if (!this.cart.Any(x => x.Id == item.Id))
+            this.cart.Add(item);
+        
+        return Task.CompletedTask;
```

Comment:
I wonder if here is a good place to add some breadcrumbs:
```suggestion
        // Informational logs become Sentry breadcrumbs automatically.
        _logger.LogInformation("Adding product to cart: {productId} - {productName}", item.Id, item.Name);
        return Task.CompletedTask;
```

Reply:
Keep in mind that MAUI buttons are already breadcrumbed (I have an x:Name on buttons that matter to grab additional context) and that I have CTMVVM commands ready to be spanned (could breadcrumb them as well).  Thoughts?

Reply:
I see. Well the goal was to "demo" that users can add crumbs too while also adding something app specific (the counts).
But if the crumbs will be too spammy than perhaps that's not ideal.

Worth a try though